homeassistant/components/conversation.py

```diff
@@ -29,20 +31,43 @@
     vol.Required(ATTR_TEXT): vol.All(cv.string, vol.Lower),
 })
 
-CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Schema({}),
-}, extra=vol.ALLOW_EXTRA)
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({
+    cv.string: vol.Schema({
+        vol.Required('keywords'): cv.string,
+        vol.Required('action'): cv.SCRIPT_SCHEMA,
+    })
+})}, extra=vol.ALLOW_EXTRA)
 
 
 def setup(hass, config):
     """Register the process service."""
-    warnings.filterwarnings('ignore', module='fuzzywuzzy')
     from fuzzywuzzy import process as fuzzyExtract
+    warnings.filterwarnings('ignore', module='fuzzywuzzy')
 
     logger = logging.getLogger(__name__)
+    config = config.get(DOMAIN, {})
+    # Right way to do this-- taken from updater.py
+
+    choices = {attrs['keywords']: script.Script(
+        hass,
+        attrs['action'],
+        name)
+        for name, attrs in config.items()} or {'': lambda: None}
+# Make sure choices isnt empty
 
     def process(service):
         """Parse text into commands."""
+
+        text = service.data[ATTR_TEXT]
+        match = fuzzyExtract.extractOne(text, choices.keys())
+        scorelimit = 60  # arbitrary value
+        logging.info('matched up text %s and found %s' % (
+            text, [match[0] if match[1] > scorelimit else 'nothing']
+            )
+        )
+        if match[1] > scorelimit:
+            return choices[match[0]].run()  # run respective script
```

Comment:
Please break this into 2 lines. One for the run and one for the return so this method is consistent in what it returns (`None`).